include/ceed/jit-tools.h

```diff
@@ -21,6 +21,7 @@
 
 CEED_EXTERN int CeedCheckFilePath(Ceed ceed, const char *source_file_path, bool *is_valid);
 CEED_EXTERN int CeedLoadSourceToBuffer(Ceed ceed, const char *source_file_path, char **buffer);
+CEED_EXTERN int CeedLoadSourceToInitializedBuffer(Ceed ceed, const char *source_file_path, char **buffer);
```

Comment:
The main change outside of the MAGMA backend is that I moved this function to the `jit-tools` header in order to easily load several kernel files before compiling.  Is this acceptable?  If not, should we move to a different header?  I'd rather not have to duplicate some common code in order to have everything in one file for each dimension's basis kernels, for example.